realm/realm-library/src/main/cpp/io_realm_internal_TableQuery.cpp

```diff
@@ -1912,3 +1909,17 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeIsEmpty
         }
     } CATCH_STD()
 }
+
+static void finalize_table_query(jlong ptr)
+{
+    TR_ENTER_PTR(ptr)
+    delete Q(ptr);
+}
+
+JNIEXPORT jlong JNICALL Java_io_realm_internal_TableQuery_nativeGetFinalizerPtr
+  (JNIEnv *, jclass)
+{
+    TR_ENTER()
+    return static_cast<jlong>(reinterpret_cast<uintptr_t>(&finalize_table_query));
```

Comment:
nope, we cannot. See http://stackoverflow.com/questions/1096341/function-pointers-casting-in-c